docs: add contract addr references and seq diagram (#317)
energywebfoundation/passport-did-auth: README.md

# Node.js Passport strategy using decentralised identifiers

## Description

> This repository consists of a Node.js Password Strategy which provides verification of the issuance of credential made regarding roles defined in an Ethereum Naming System (ENS).

Reviewer: ```suggestion
This repository consists of a Node.js Password Strategy which authenticates based on roles defined in Energy Web Foundation Ethereum namespace
```

## Sequence diagram

 ```mermaid
 sequenceDiagram
autonumber
    participant C as Client
    participant LS as LoginStrategy
    participant CV as ClaimVerifier
    participant IV as IssuerVerification

>     participant CS as Cache Server

Reviewer: ```suggestion
    participant CS as SSI-HUB
```

    participant IPFS
    participant DR as DomainReader
rect rgb(200, 255, 255)
    Note right of C: Initialisation

>     C->C: Client sets the login strategy options and initialises LoginStrategy

Reviewer: Arrow is better correspond to call
```suggestion
    C->>C: Client sets the login strategy options and initialises LoginStrategy
```

    Note right of C: Validation Call
    C->LS: LoginStrategy.validate(token, payload)
end
rect rgb(255, 220, 255)
    Note right of LS: Signature and role validation

>     LS->LS: Verifies token and payload (signatures)

Reviewer: I think we can specify that token signature is verified by ProofVerifier?
```suggestion
    LS->>PV: Authenticate token issuer
```

Author: @JGiter What is PV?

Reviewer: ProofVerifier from ew-did-registry
